include/glow/Support/Error.h

```diff
@@ -280,43 +284,44 @@ class GlowErrorValue final {
   /// line number the ErrorValue was created on as well as the message and/or
   /// error code the ErrorValue was created with.
   template <typename StreamT> void log(StreamT &os) const {
-    os << "location: " << fileName_ << ":" << lineNumber_;
     if (ec_ != ErrorCode::UNKNOWN) {
-      os << " error code: " << errorCodeToString(ec_);
+      os << "\nError code: " << errorCodeToString(ec_);
     }
     if (!message_.empty()) {
-      os << " message: " << message_;
+      os << "\nError message: " << message_;
+    }
+    os << "\nError return stack:\n";
+    for (const auto &p : stack_) {
+      os << p.first << ":" << p.second << "\n";
     }
   }
 
+  /// Add \p filename and \p lineNumber to the ErrorValue's stack for logging.
+  void addToStack(const std::string &fileName, size_t lineNumber) {
+    stack_.push_back({fileName, lineNumber});
+  }
+
   std::string logToString() const;
 
-  GlowErrorValue(const char *fileName, size_t lineNumber, std::string message,
-                 ErrorCode ec)
-      : lineNumber_(lineNumber), fileName_(fileName), message_(message),
-        ec_(ec) {}
+  GlowErrorValue(std::string message, ErrorCode ec)
+      : message_(message), ec_(ec) {}
 
-  GlowErrorValue(const char *fileName, size_t lineNumber, ErrorCode ec,
-                 std::string message)
-      : lineNumber_(lineNumber), fileName_(fileName), message_(message),
-        ec_(ec) {}
+  GlowErrorValue(ErrorCode ec, std::string message)
+      : message_(message), ec_(ec) {}
 
-  GlowErrorValue(const char *fileName, size_t lineNumber, ErrorCode ec)
-      : lineNumber_(lineNumber), fileName_(fileName), ec_(ec) {}
+  GlowErrorValue(ErrorCode ec) : ec_(ec) {}
 
-  GlowErrorValue(const char *fileName, size_t lineNumber, std::string message)
-      : lineNumber_(lineNumber), fileName_(fileName), message_(message) {}
+  GlowErrorValue(std::string message) : message_(message) {}
 
 private:
   /// Convert ErrorCode values to string.
   static std::string errorCodeToString(const ErrorCode &ec);
 
-  /// The line number the error was generated on.
-  size_t lineNumber_;
-  /// The name of the file the error was generated in.
-  std::string fileName_;
   /// Optional message associated with the error.
   std::string message_;
+  /// The filename and line number of all places that created, forwarded, and
+  /// destroyed the ErrorValue.
+  std::vector<std::pair<std::string, size_t>> stack_;
```

Comment:
Quick question: Do we need to be concerned about the overhead of growing this vector and performing dynamic memory allocations? Or is it very low and we don't care?

Reply:
It's a good question, especially given that is code executing on error path.
I think it should be sufficiently small that this doesn't matter but if you know of a case where this could be an issue but it should actually be possible to avoid allocations here if it becomes a problem.

Reply:
As to the overheads specifically, allocations only happens when an Error actually occurs